### PR TITLE
Change constructors visibility for easier extensibility

### PR DESCRIPTION
**Pull request by edgars-pivovarenoks**

### CODE REVIEW

File: src/Fluent.Calculations.Primitives/Tag.cs [hunk 1]
```diff
@@ -2,6 +2,8 @@
 {
     public class Tag
     {
-        public string Name { get; } = "Undefined";
+        private Tag() { }
+        public static Tag Create(string name) => new Tag { Name = name };
+        public string Name { get; private set; } = "Undefined";
     }
```
Comment (ArturKarbone) [inline src/Fluent.Calculations.Primitives/Tag.cs:8]: @edgars-pivovarenoks  I suggest stick to immutability principles here
```csharp
 public required string Name { get; } = "Undefined";
```

File: src/Fluent.Calculations.Primitives/Value.cs [hunk 2]
```diff
@@ -8,6 +8,15 @@ public abstract class Value : IValue, IName
...
     private Value() { }
 
+    public Value(Value value)
+    {
+        Name = value.Name;
+        Expression = value.Expression;
+        PrimitiveValue = value.PrimitiveValue;
+        IsConstant = value.IsConstant;
+        Tags = value.Tags;
+    }
```
Comment (ArturKarbone) [inline src/Fluent.Calculations.Primitives/Value.cs:18]: @edgars-pivovarenoks what if make it immutable:
```csharp
public Copy() => new Value(this):
protected Value(Value value)
    {
        Name = value.Name;
        Expression = value.Expression;
        PrimitiveValue = value.PrimitiveValue;
        IsConstant = value.IsConstant;
        Tags = value.Tags;
    }
```

**Review by ArturKarbone — CHANGES_REQUESTED** [review]

File: src/Fluent.Calculations.Primitives/Value.cs (hunk 2 above)
Comment (edgars-pivovarenoks) [reply src/Fluent.Calculations.Primitives/Value.cs:18]: @ArturKarbone  but it is immutable. This is a constructor. Copy method can't help with inhertiance.

File: src/Fluent.Calculations.Primitives/Tag.cs (hunk 1 above)
Comment (edgars-pivovarenoks) [reply src/Fluent.Calculations.Primitives/Tag.cs:8]: It's basically a plug to compiler warning. The class is immutable. Let's discuss offline.

**Review by ArturKarbone — APPROVED** [review]

**Review by ArturKarbone — APPROVED** [review]

**Pull request merged**